### DISCUSSION TITLE
Questionset report not working - Content/state/update API is not getting called after compliting and rating the assesment course.

### DISCUSSION THREAD

**Discussion by paritshivani** [Sunbird-Lern/Community#152] [Installation] [answered]
Hi folks,
Hope you are doing great!

We are working on question set reports. As per our understanding, Content state gets updated  to 2 once we complete the assessment and give rating to content. We observed that content state is getting updated to status 2 but still report has no data in it.

I have following doubts:
1. Do the rating content required for state update and for report?
2. Do we need anything else to be done after content compleation to update report data?
3. What parameters need to be checked 

@coolbung, @reshmi-nair, @sravani-64, @Nithin-ns24 , @gandham-santhosh, @kumarks1122

**Comment by sravani-64** [comment]
Hi team, the generated report has no data and we got to know that content state update API has to be called for generating this data. Can someone please help us understand the scenario? 
![image](https://github.com/Sunbird-Lern/Community/assets/133327173/26e79f0c-a76f-4d31-b3da-058fba3bcd0e)

**Reply by Jayaprakash8887** [reply]
@sravani-64 Hi, has the course been marked as "Course Assessment" type? Please share the course metadata

**Reply by paritshivani** [reply]
here are the course details:
`{
  "content": {
    "ownershipType": [
      "createdBy"
    ],
    "publish_type": "public",
    "copyright": "Org1",
    "se_gradeLevelIds": [
      "project_gradelevel_knowledge",
      "project_gradelevel_mission",
      "project_gradelevel_process",
      "project_gradelevel_program",
      "project_gradelevel_software",
      "project_gradelevel_solution"
    ],
    "keywords": [
      "course"
    ],
    "subject": [
      "How To"
    ],
    "targetMediumIds": [
      "project_medium_english",
      "project_medium_hindi",
      "project_medium_marathi",
      "project_medium_telugu"
    ],
    "channel": "0130701891041689600",
    "downloadUrl": "https://......blob.core.windows.net/contents/content/do_1138565902748794881240/8th-aug-course-test-two_1691478487112_do_1138565902748794881240_1_SPINE.ecar",
    "organisation": [
      "Org1"
    ],
    "language": [
      "English"
    ],
    "mimeType": "application/vnd.ekstep.content-collection",
    "variants": "{\"spine\":{\"ecarUrl\":\"https://......blob.core.windows.net/contents/content/do_1138565902748794881240/8th-aug-course-test-two_1691478487112_do_1138565902748794881240_1_SPINE.ecar\",\"size\":\"5718\"},\"online\":{\"ecarUrl\":\"https://......blob.core.windows.net/contents/content/do_1138565902748794881240/8th-aug-course-test-two_1691478487162_do_1138565902748794881240_1_ONLINE.ecar\",\"size\":\"5717\"}}",
    "body": null,
    "leafNodes": [
      "do_113833407416852480148"
    ],
    "targetGradeLevelIds": [
      "project_gradelevel_knowledge",
      "project_gradelevel_mission",
      "project_gradelevel_process",
      "project_gradelevel_program",
      "project_gradelevel_software",
      "project_gradelevel_solution"
    ],
    "objectType": "Content",
    "se_mediums": [
      "English",
      "Hindi",
      "Marathi",
      "Telugu"
    ],
    "appIcon": "",
    "children": [
      {
        "lastStatusChangedOn": "2023-08-08T07:05:58.809+0000",
        "parent": "do_1138565902748794881240",
        "children": [
          {
            "copyright": "project",
            "lastStatusChangedOn": "2023-07-06T13:09:22.261+0000",
            "parent": "do_1138565907153633281241",
            "organisation": [
              "project"
            ],
            "mediaType": "content",
            "name": "6th July Course Assessment Test one",
            "discussionForum": "{\"enabled\":\"No\"}",
            "createdOn": "2023-07-06T12:59:26.317+0000",
            "createdFor": [
              "0130152125009674241"
            ],
            "channel": "0130152125009674241",
            "assets": [],
            "plugins": "[{\"identifier\":\"org.ekstep.stage\",\"semanticVersion\":\"1.0\"},{\"identifier\":\"org.ekstep.questionset\",\"semanticVersion\":\"1.0\"},{\"identifier\":\"org.ekstep.navigation\",\"semanticVersion\":\"1.0\"},{\"identifier\":\"org.ekstep.questionset.quiz\",\"semanticVersion\":\"1.0\"},{\"identifier\":\"org.ekstep.iterator\",\"semanticVersion\":\"1.0\"},{\"identifier\":\"org.ekstep.questionunit\",\"semanticVersion\":\"1.2\"},{\"identifier\":\"org.ekstep.keyboard\",\"semanticVersion\":\"1.1\"},{\"identifier\":\"org.ekstep.questionunit.ftb\",\"semanticVersion\":\"1.1\"},{\"identifier\":\"org.ekstep.summary\",\"semanticVersion\":\"1.0\"}]",
            "lastUpdatedOn": "2023-07-06T13:09:22.261+0000",
            "subject": [
              "Standard Operating Procedure"
            ],
            "size": 110715,
            "editorState": "{\"plugin\":{\"noOfExtPlugins\":12,\"extPlugins\":[{\"plugin\":\"org.ekstep.contenteditorfunctions\",\"version\":\"1.2\"},{\"plugin\":\"org.ekstep.keyboardshortcuts\",\"version\":\"1.0\"},{\"plugin\":\"org.ekstep.richtext\",\"version\":\"1.0\"},{\"plugin\":\"org.ekstep.iterator\",\"version\":\"1.0\"},{\"plugin\":\"org.ekstep.navigation\",\"version\":\"1.0\"},{\"plugin\":\"org.ekstep.reviewercomments\",\"version\":\"1.0\"},{\"plugin\":\"org.ekstep.questionunit.mtf\",\"version\":\"1.2\"},{\"plugin\":\"org.ekstep.questionunit.mcq\",\"version\":\"1.3\"},{\"plugin\":\"org.ekstep.keyboard\",\"version\":\"1.1\"},{\"plugin\":\"org.ekstep.questionunit.reorder\",\"version\":\"1.1\"},{\"plugin\":\"org.ekstep.questionunit.sequence\",\"version\":\"1.1\"},{\"plugin\":\"org.ekstep.questionunit.ftb\",\"version\":\"1.1\"}]},\"stage\":{\"noOfStages\":1,\"currentStage\":\"124ac35d-2701-4d74-83a5-a26d01b18875\",\"selectedPluginObject\":\"7184b233-6e6b-4422-9462-b6c7c02c05e2\"},\"sidebar\":{\"selectedMenu\":\"settings\"}}",
            "streamingUrl": "https://......blob.core.windows.net/contents/content/ecml/do_113833407416852480148-latest",
            "identifier": "do_113833407416852480148",
            "description": "Enter description for Assessment",
            "gradeLevel": [
              "Program"
            ],
            "ownershipType": [
              "createdBy"
            ],
            "compatibilityLevel": 1,
            "audience": [
              "Student"
            ],
            "se_boards": [
              "Water"
            ],
            "os": [
              "All"
            ],
            "cloudStorageKey": "content/do_113833407416852480148/artifact/1688648959420_do_113833407416852480148.zip",
            "primaryCategory": "Course Assessment",
            "se_mediums": [
              "English"
            ],
            "downloadUrl": "https://......blob.core.windows.net/contents/content/do_113833407416852480148/6th-july-course-assessment-test-one_1688648962146_do_113833407416852480148_1.ecar",
            "se_subjects": [
              "Standard Operating Procedure"
            ],
            "lockKey": "9602b048-d437-40fa-bba4-8eccc02d216a",
            "medium": [
              "English"
            ],
            "framework": "project",
            "creator": "Vishal RC",
            "versionKey": "1688648878436",
            "mimeType": "application/vnd.ekstep.ecml-archive",
            "code": "org.sunbird.lUmEQn",
            "license": "CC BY 4.0",
            "maxAttempts": 3,
            "version": 2,
            "prevStatus": "Processing",
            "contentType": "SelfAssess",
            "prevState": "Review",
            "language": [
              "English"
            ],
            "board": "Water",
            "totalQuestions": 1,
            "lastPublishedOn": "2023-07-06T13:09:20.443+0000",
            "totalScore": 1,
            "objectType": "Content",
            "lastUpdatedBy": "1093a032-fb64-4e13-bf3b-4fb169032aae",
            "status": "Live",
            "pragma": [],
            "createdBy": "1093a032-fb64-4e13-bf3b-4fb169032aae",
            "dialcodeRequired": "No",
            "lastSubmittedOn": "2023-07-06T13:07:58.399+0000",
            "interceptionPoints": "{}",
            "idealScreenSize": "normal",
            "contentEncoding": "gzip",
            "depth": 2,
            "consumerId": "4c1d6bc8-8be3-4c9e-96e8-51a26738e76e",
            "lastPublishedBy": "767ddff1-eb21-4835-b498-0b8aac15f33f",
            "se_gradeLevels": [
              "Program"
            ],
            "osId": "org.ekstep.quiz.app",
            "se_FWIds": [
              "project"
            ],
            "contentDisposition": "inline",
            "previewUrl": "https://......blob.core.windows.net/contents/content/ecml/do_113833407416852480148-latest",
            "artifactUrl": "https://......blob.core.windows.net/contents/content/do_113833407416852480148/artifact/1688648959420_do_113833407416852480148.zip",
            "visibility": "Default",
            "credentials": "{\"enabled\":\"No\"}",
            "variants": "{\"full\":{\"ecarUrl\":\"https://......blob.core.windows.net/contents/content/do_113833407416852480148/6th-july-course-assessment-test-one_1688648962146_do_113833407416852480148_1.ecar\",\"size\":\"108284\"},\"spine\":{\"ecarUrl\":\"https://......blob.core.windows.net/contents/content/do_113833407416852480148/6th-july-course-assessment-test-one_1688648962188_do_113833407416852480148_1_SPINE.ecar\",\"size\":\"4404\"}}",
            "index": 1,
            "pkgVersion": 1,
            "idealScreenDensity": "hdpi"
          }
        ],
        "mediaType": "content",
        "name": "Course Unit",
        "discussionForum": {
          "enabled": "No"
        },
        "createdOn": "2023-08-08T07:05:58.809+0000",
        "channel": "0130701891041689600",
        "generateDIALCodes": "No",
        "lastUpdatedOn": "2023-08-08T07:07:22.580+0000",
        "identifier": "do_1138565907153633281241",
        "ownershipType": [
          "createdBy"
        ],
        "compatibilityLevel": 1,
        "audience": [
          "Student"
        ],
        "trackable": {
          "enabled": "No",
          "autoBatch": "No"
        },
        "os": [
          "All"
        ],
        "primaryCategory": "Course Unit",
        "languageCode": [
          "en"
        ],
        "downloadUrl": "https://......blob.core.windows.net/contents/content/do_1138565902748794881240/8th-aug-course-test-two_1691478487112_do_1138565902748794881240_1_SPINE.ecar",
        "attributions": [],
        "versionKey": "1691478358809",
        "mimeType": "application/vnd.ekstep.content-collection",
        "code": "fe38df62-da07-82dc-1732-72e72e554869",
        "license": "CC BY 4.0",
        "leafNodes": [
          "do_113833407416852480148"
        ],
        "version": 2,
        "contentType": "CourseUnit",
        "language": [
          "English"
        ],
        "lastPublishedOn": "2023-08-08T07:08:07.013+0000",
        "objectType": "Content",
        "status": "Live",
        "dialcodeRequired": "No",
        "idealScreenSize": "normal",
        "contentEncoding": "gzip",
        "leafNodesCount": 1,
        "depth": 1,
        "osId": "org.ekstep.launcher",
        "contentDisposition": "inline",
        "visibility": "Parent",
        "credentials": {
          "enabled": "No"
        },
        "variants": "{\"spine\":{\"ecarUrl\":\"https://......blob.core.windows.net/contents/content/do_1138565902748794881240/8th-aug-course-test-two_1691478487112_do_1138565902748794881240_1_SPINE.ecar\",\"size\":\"5718\"},\"online\":{\"ecarUrl\":\"https://......blob.core.windows.net/contents/content/do_1138565902748794881240/8th-aug-course-test-two_1691478487162_do_1138565902748794881240_1_ONLINE.ecar\",\"size\":\"5717\"}}",
        "index": 1,
        "pkgVersion": 1,
        "idealScreenDensity": "hdpi"
      }
    ],
    "primaryCategory": "Course",
    "contentEncoding": "gzip",
    "lockKey": "9ddba155-a2ba-499e-92ea-2beffdd613d8",
    "generateDIALCodes": "No",
    "totalCompressedSize": 110715,
    "mimeTypesCount": "{\"application/vnd.ekstep.ecml-archive\":1,\"application/vnd.ekstep.content-collection\":1}",
    "sYS_INTERNAL_LAST_UPDATED_ON": "2023-08-08T07:08:07.111+0000",
    "contentType": "Course",
    "se_gradeLevels": [
      "Knowledge",
      "Mission",
      "Process",
      "Program",
      "Software",
      "Solution"
    ],
    "trackable": {
      "autoBatch": "Yes",
      "enabled": "Yes"
    },
    "identifier": "do_1138565902748794881240",
    "audience": [
      "Student"
    ],
    "se_boardIds": [
      "project_board_it"
    ],
    "subjectIds": [
      "project_subject_howto"
    ],
    "toc_url": "https://......blob.core.windows.net/contents/content/do_1138565902748794881240/artifact/do_1138565902748794881240_toc.json",
    "visibility": "Default",
    "contentTypesCount": "{\"SelfAssess\":1,\"CourseUnit\":1}",
    "consumerId": "4c1d6bc8-8be3-4c9e-96e8-51a26738e76e",
    "childNodes": [
      "do_113833407416852480148",
      "do_1138565907153633281241"
    ],
    "discussionForum": {
      "enabled": "No"
    },
    "mediaType": "content",
    "osId": "org.ekstep.quiz.app",
    "lastPublishedBy": "6ba0f0b5-67e6-42cf-850c-f426e0a7ddc4",
    "version": 2,
    "se_subjects": [
      "How To"
    ],
    "license": "CC BY 4.0",
    "size": 5718,
    "lastPublishedOn": "2023-08-08T07:08:07.013+0000",
    "name": "8th Aug Course Test two",
    "attributions": [],
    "targetBoardIds": [
      "project_board_it"
    ],
    "status": "Live",
    "code": "org.sunbird.N85BUj",
    "publishError": null,
    "credentials": {
      "enabled": "Yes"
    },
    "description": "Enter description for Course",
    "idealScreenSize": "normal",
    "createdOn": "2023-08-08T07:05:05.041+0000",
    "batches": [
      {
        "createdFor": [
          "0130701891041689600"
        ],
        "endDate": "2023-09-30",
        "name": "Tekdi Test Batch",
        "batchId": "01385660196594483215",
        "enrollmentType": "open",
        "enrollmentEndDate": "2023-09-30",
        "startDate": "2023-08-08",
        "status": 1
      }
    ],
    "se_boards": [
      "IT"
    ],
    "targetSubjectIds": [
      "project_subject_bestpractices",
      "project_subject_firefightingknowledge",
      "project_subject_howto",
      "project_subject_innovation",
      "project_subject_overview",
      "project_subject_routinejobrequirement",
      "project_subject_standardoperatingprocedure"
    ],
    "se_mediumIds": [
      "project_medium_english",
      "project_medium_hindi",
      "project_medium_marathi",
      "project_medium_telugu"
    ],
    "copyrightYear": 2022,
    "contentDisposition": "inline",
    "lastUpdatedOn": "2023-08-08T07:07:22.580+0000",
    "dialcodeRequired": "No",
    "lastStatusChangedOn": "2023-08-08T07:05:05.041+0000",
    "createdFor": [
      "0130701891041689600"
    ],
    "creator": "Org1_creator  User2",
    "os": [
      "All"
    ],
    "flagReasons": null,
    "se_subjectIds": [
      "project_subject_howto",
      "project_subject_bestpractices",
      "project_subject_firefightingknowledge",
      "project_subject_innovation",
      "project_subject_overview",
      "project_subject_routinejobrequirement",
      "project_subject_standardoperatingprocedure"
    ],
    "se_FWIds": [
      "project"
    ],
    "targetFWIds": [
      "project"
    ],
    "pkgVersion": 1,
    "versionKey": "1691478442580",
    "idealScreenDensity": "hdpi",
    "framework": "project",
    "dialcodes": null,
    "depth": 0,
    "s3Key": "content/do_1138565902748794881240/artifact/do_1138565902748794881240_toc.json",
    "lastSubmittedOn": "2023-08-08T07:07:22.522+0000",
    "createdBy": "86422ccb-2c90-4e72-a842-3c97a32a2806",
    "compatibilityLevel": 4,
    "leafNodesCount": 1,
    "userConsent": "Yes",
    "resourceType": "Course",
    "orgDetails": {
      "email": null,
      "orgName": "Org1"
    },
    "licenseDetails": {
      "name": "CC BY 4.0",
      "url": "https://creativecommons.org/licenses/by/4.0/legalcode",
      "description": "This is the standard license of any Youtube content"
    }
  }
}`
@Jayaprakash8887, Please check

**Comment by sravani-64** [comment]
@Jayaprakash8887 can you pls look into the metadata shared by Shivani?

**Reply by Jayaprakash8887** [reply]
@sravani-64 

Do the rating content required for state update and for report? - Answer: NO
Do we need anything else to be done after content compleation to update report data? - Answer: Ideally Content state API should be triggered automatically from the consumption portal. Please check if the same is getting triggered via network tab

**Reply by Jayaprakash8887** [reply]
You can also refer to https://lern.sunbird.org/learn/product-and-developer-guide/lms-service/lms-flink-jobs and https://lern.sunbird.org/learn/product-and-developer-guide/lms-service/reports for better understanding

**Reply by Jayaprakash8887** [reply]
https://lern.sunbird.org/learn/product-and-developer-guide/lms-service/reports/on-demand-exhaust/response-exhaust

**Comment by mohitga** [comment]
@paritshivani 
Hope you are doing good. 
Let us know if the issue is resolved using the solution thread suggested above?
Thanks.

**Comment by paritshivani** [comment]
@mohitga , @Jayaprakash8887 , @sravani-64 
Content state update is getting called two times while consuming content

 first update:
Curl:
`curl 'https://devnulp.niua.org/content/course/v1/content/state/update' \
  -X 'PATCH' \
  -H 'Accept: application/json' \
  -H 'Accept-Language: en-GB,en-US;q=0.9,en;q=0.8' \
  -H 'Connection: keep-alive' \
  -H 'Content-Type: application/json' \
  -H 'Cookie: _gid=GA1.2.740725679.1692596556; _ga=GA1.1.2024172102.1678947259; connect.sid=s%3AWEL0ACMeLtSbjkhI6Jdd-_cu2PiNpNDv.YQ3w%2BikXVb3bEZr4r3bneLqs2BF91vB%2F25gyzOJQgTo; _ga_QH3SHT9MTG=GS1.1.1692596555.239.0.1692596557.0.0.0; _ga_EJDFKF9L1X=GS1.1.1692596555.179.0.1692596557.0.0.0' \
  -H 'Origin: https://devnulp.niua.org' \
  -H 'Referer: https://devnulp.niua.org/learn/course/play/do_1138565907153633281241?batchId=01385660196594483215&courseId=do_1138565902748794881240&courseName=8th%20Aug%20Course%20Test%20two&selectedContent=do_113833407416852480148' \
  -H 'Sec-Fetch-Dest: empty' \
  -H 'Sec-Fetch-Mode: cors' \
  -H 'Sec-Fetch-Site: same-origin' \
  -H 'User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/112.0.0.0 Safari/537.36' \
  -H 'X-App-Id: dev.nulp.portal' \
  -H 'X-App-Version: 4.10.0' \
  -H 'X-Channel-Id: 0130701891041689600' \
  -H 'X-Device-ID: 8ed7b40ed64564ee1139ba87ab06cca3' \
  -H 'X-Org-code: 0130701891041689600' \
  -H 'X-Request-ID: c4577629-2a00-35b2-c6f3-bd2163e3c937' \
  -H 'X-Session-ID: WEL0ACMeLtSbjkhI6Jdd-_cu2PiNpNDv' \
  -H 'X-Source: web' \
  -H 'X-User-ID: 5d757783-a86a-40cd-a814-1b6a16d37cb6' \
  -H 'X-msgid: c4577629-2a00-35b2-c6f3-bd2163e3c937' \
  -H 'sec-ch-ua: "Chromium";v="112", "Google Chrome";v="112", "Not:A-Brand";v="99"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "Linux"' \
  -H 'ts: 2023-08-21T11:13:59+05:30' \
  --data-raw '{"request":{"userId":"5d757783-a86a-40cd-a814-1b6a16d37cb6","contents":[{"contentId":"do_113833407416852480148","batchId":"01385660196594483215","status":2,"courseId":"do_1138565902748794881240","lastAccessTime":"2023-08-21 11:13:59:088+0530"}]}}' \
  --compressed`
Response:
`{
  "id": "api.content.state.update",
  "ver": "v1",
  "ts": "2023-08-21 05:43:59:130+0000",
  "params": {
    "resmsgid": null,
    "msgid": "c4577629-2a00-35b2-c6f3-bd2163e3c937",
    "err": null,
    "status": "success",
    "errmsg": null
  },
  "responseCode": "OK",
  "result": {
    "do_113833407416852480148": "SUCCESS"
  }
}`



 second update:
curl:
`curl 'https://site/content/course/v1/content/state/update' \
  -X 'PATCH' \
  -H 'Accept: application/json' \
  -H 'Accept-Language: en-GB,en-US;q=0.9,en;q=0.8' \
  -H 'Connection: keep-alive' \
  -H 'Content-Type: application/json' \
  -H 'Cookie: _gid=GA1.2.740725679.1692596556; _ga=GA1.1.2024172102.1678947259; connect.sid=s%3AWEL0ACMeLtSbjkhI6Jdd-_cu2PiNpNDv.YQ3w%2BikXVb3bEZr4r3bneLqs2BF91vB%2F25gyzOJQgTo; _ga_QH3SHT9MTG=GS1.1.1692596555.239.0.1692596557.0.0.0; _ga_EJDFKF9L1X=GS1.1.1692596555.179.0.1692596557.0.0.0' \
  -H 'Origin: https://site' \
  -H 'Referer: https://devnulp.niua.org/learn/course/play/do_1138565907153633281241?batchId=01385660196594483215&courseId=do_1138565902748794881240&courseName=8th%20Aug%20Course%20Test%20two&selectedContent=do_113833407416852480148' \
  -H 'Sec-Fetch-Dest: empty' \
  -H 'Sec-Fetch-Mode: cors' \
  -H 'Sec-Fetch-Site: same-origin' \
  -H 'User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/112.0.0.0 Safari/537.36' \
  -H 'X-App-Id: dev.nulp.portal' \
  -H 'X-App-Version: 4.10.0' \
  -H 'X-Channel-Id: 0130701891041689600' \
  -H 'X-Device-ID: 8ed7b40ed64564ee1139ba87ab06cca3' \
  -H 'X-Org-code: 0130701891041689600' \
  -H 'X-Request-ID: 614eabff-b3ab-cceb-ae59-28fc2a2669fe' \
  -H 'X-Session-ID: WEL0ACMeLtSbjkhI6Jdd-_cu2PiNpNDv' \
  -H 'X-Source: web' \
  -H 'X-User-ID: 5d757783-a86a-40cd-a814-1b6a16d37cb6' \
  -H 'X-msgid: 614eabff-b3ab-cceb-ae59-28fc2a2669fe' \
  -H 'sec-ch-ua: "Chromium";v="112", "Google Chrome";v="112", "Not:A-Brand";v="99"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "Linux"' \
  -H 'ts: 2023-08-21T11:13:59+05:30' \
  --data-raw '{"request":{"userId":"5d757783-a86a-40cd-a814-1b6a16d37cb6","contents":[{"contentId":"do_113833407416852480148","batchId":"01385660196594483215","status":2,"courseId":"do_1138565902748794881240","lastAccessTime":"2023-08-21 11:13:59:189+0530"}],"assessments":[{"assessmentTs":1692596604774,"batchId":"01385660196594483215","courseId":"do_1138565902748794881240","userId":"5d757783-a86a-40cd-a814-1b6a16d37cb6","attemptId":"6d4a23a635fa584826277b0200a21880","contentId":"do_113833407416852480148","events":[{"eid":"ASSESS","ets":1692596632174,"ver":"3.1","mid":"ASSESS:78694fc4f49d0d57349a791304e73f6f","actor":{"id":"5d757783-a86a-40cd-a814-1b6a16d37cb6","type":"User"},"context":{"channel":"0130701891041689600","pdata":{"id":"dev.nulp.portal","ver":"4.10.0","pid":"sunbird-portal.contentplayer"},"env":"contentplayer","sid":"WEL0ACMeLtSbjkhI6Jdd-_cu2PiNpNDv","did":"8ed7b40ed64564ee1139ba87ab06cca3","cdata":[{"id":"do_1138565907153633281241","type":"course"},{"type":"batch","id":"01385660196594483215"},{"id":"3a1e232ce5b6fdb59d60fe90e1da2226","type":"ContentSession"},{"id":"1477ba4e8444e82674411529cdb04992","type":"PlaySession"}],"rollup":{"l1":"0130701891041689600"}},"object":{"id":"do_113833407416852480148","type":"Content","ver":"1","rollup":{"l1":"do_1138565907153633281241","l2":"do_113833407416852480148"}},"tags":["0130701891041689600"],"edata":{"item":{"id":"do_113717285658771456171","maxscore":1,"type":"ftb","exlength":0,"params":[{"1":"{\"text\":\"p\"}"},{"2":"{\"text\":\"p\"}"},{"eval":"order"}],"uri":"","title":"FILL IN THE BLANK\n\n \n\nTHE ____ AND ____.\n","mmc":[],"mc":[],"desc":"Its MTF"},"index":1,"pass":"No","score":0,"resvalues":[{"1":"{\"text\":\"PMK \"}"}],"duration":25}}]}]}}' \
  --compressed`

Response:
`{
  "id": "api.content.state.update",
  "ver": "v1",
  "ts": "2023-08-21 05:43:59:217+0000",
  "params": {
    "resmsgid": null,
    "msgid": "614eabff-b3ab-cceb-ae59-28fc2a2669fe",
    "err": null,
    "status": "success",
    "errmsg": null
  },
  "responseCode": "OK",
  "result": {
    "01385660196594483215": "SUCCESS",
    "do_113833407416852480148": "SUCCESS"
  }
}`

**Reply by mohitga** [reply]
@BharathwajShankar  please have a look at it.

**Reply by Jayaprakash8887** [reply]
@paritshivani APIs triggered seems to be fine. Next set of verification steps is, as mentioned in this diagram "https://1497159047-files.gitbook.io/~/files/v0/b/gitbook-x-prod.appspot.com/o/spaces%2F4ZKyfmmhMWpPkD6iYvKF%2Fuploads%2FDgQfqjP2sc6TQgQG64xV%2FUntitled%20Diagram-V2%20LERN%20-%20CERTIFICATE%20GENERATION%20FLOW.drawio.png?alt=media&token=54df7c55-5ea1-4da3-879c-61ae2c18dfee"  , please check if the event is getting pushed to "assessment-aggregator" flink job kafka topic. Please check if the flink job is working fine.

**Reply by sravani-64** [reply]
@Jayaprakash8887 
1. For the users with teacher role and framework defined, assessment aggregator table from cassandra db is giving data and generated response exhaust report also has data 
![image](https://github.com/Sunbird-Lern/Community/assets/133327173/a3318bbe-5f28-4f74-80db-6d866b66da84)
2. For the users with same teacher role and no framework, data is missing from assessment aggregator table and hence generated response exhaust report also has no data
![image](https://github.com/Sunbird-Lern/Community/assets/133327173/acb4175b-6163-4f2d-9608-dc784647ae5d)
Can you pls help us know if there is any criteria for generating the data in assessment aggregator? Thank you

**Reply by Jayaprakash8887** [reply]
@sravani-64 pls run the job locally in debug mode and check the deviations happening. https://lern.sunbird.org/learn/product-and-developer-guide/lms-service/lms-flink-jobs/assessment-aggregator

**Reply by sravani-64** [reply]
@Jayaprakash8887 I have checked the events in kafka and found that events are present in dev.issue.certificate.request topic for both do_ids do_1138565919174410241243 and do_1138565902748794881240 (different courses)
![image](https://github.com/Sunbird-Lern/Community/assets/133327173/bd48736b-c001-4196-a4cd-06415351fe4a)
**But couldn't find the data in assessment aggregator table with do_id do_1138565919174410241243**
![image](https://github.com/Sunbird-Lern/Community/assets/133327173/9ca4183a-5531-4a2d-8edc-a5bc2c82c3b3)

I have triggered the response exhaust cronjob in debug mode but couldn't find any information

**Reply by Jayaprakash8887** [reply]
@sravani-64 pls run the job locally in debug mode and check the deviations happening. https://lern.sunbird.org/learn/product-and-developer-guide/lms-service/lms-flink-jobs/assessment-aggregator

**Reply by Jayaprakash8887** [reply]
@sravani-64 Hope the issue is resolved. Kindly mark the thread as answered if there are no further queries.

**Reply by sravani-64** [reply]
@Jayaprakash8887 we are still facing the issue, couldn't find any clues even after running the job in debug mode

**Comment by BharathwajShankar** [comment]
@paritshivani can you please let me know what is the exact issue you are facing or what is your expectation here So that I can help you better.

**Reply by mohitga** [reply]
@paritshivani  please leat us know if you are still facing the issue?

**Reply by sravani-64** [reply]
yes @mohitga  we are not able to find data after generating questionset report. I have executed the job in debug mode but couldn't find any clues

**Reply by AmiableAnil** [reply]
@sravani-64 @paritshivani can we connect at 2:30PM today? Please let me know, so that I'll share the link here.

**Reply by sravani-64** [reply]
@AmiableAnil yes we can connect

**Reply by AmiableAnil** [reply]
@sravani-64 Please join here at 2:30PM [Click here to join the meeting](https://teams.microsoft.com/l/meetup-join/19%3ameeting_OWI4Mjg2MGItZDlmNC00ZjJjLWEwOTItNmVjYTdmMGQwZWQx%40thread.v2/0?context=%7b%22Tid%22%3a%22d2f80462-bf28-4055-8ef3-085d8e4b2ed0%22%2c%22Oid%22%3a%222d4e8def-ebcf-474e-a0ac-63e56966a02a%22%7d)

**Reply by sravani-64** [reply, marked as answer]
@AmiableAnil , @kumarks1122 thank you for connecting with us. Adding the summary of our discussion here. 
The content state update API called from UI has no assess events generated. So, dev team will check this issue from front end and we will get back to you, if we face the issue even after fixing the UI assess events issue.

**Reply by Jayaprakash8887** [reply]
@sravani-64 Please mark the thread as answered if there are no further queries.
